notebooks/worst_case_probs.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+import numba
+from robupy.auxiliary import get_worst_case_probs
+from ruspy.estimation.estimation_cost_parameters import create_transition_matrix
+
+
+@numba.jit(nopython=True)
+def create_worst_trans_mat(trans_mat, v, rho):
+    num_states = trans_mat.shape[0]
+    worst_trans_mat = np.zeros(shape=(num_states, num_states), dtype=np.float64)
+    for s in range(num_states):
+        ind_non_zero = np.nonzero(trans_mat[s, :])[0]
+        p_min = np.amin(ind_non_zero)
+        p_max = np.amax(ind_non_zero)
+        p = trans_mat[s, p_min: p_max + 1]
+        v_intern = v[p_min: p_max + 1]
+        worst_trans_mat[s, p_min: p_max + 1] = get_worst_case_probs(
+            v_intern, p, rho, is_cost=False
+        )
+    return worst_trans_mat
+
+
+@numba.jit(nopython=True)
+def calc_fixp_worst(
+    num_states, p_ml, costs, beta, rho, threshold=1e-8, max_it=100000000
+):
+    ev = np.zeros(num_states)
+    worst_trans_mat = trans_mat = create_transition_matrix(num_states, p_ml)
+    ev_new = np.dot(trans_mat, np.log(np.sum(np.exp(-costs), axis=1)))
+    while (np.max(np.abs(ev_new - ev)) > threshold) & (max_it != 0):
+        ev = ev_new
+        maint_value = beta * ev - costs[:, 0]
+        repl_value = beta * ev[0] - costs[0, 1] - costs[0, 0]
+        ev_min = maint_value[0]
```

Comment:
This is the recentering in Equation 3.20 in Rust (200), please give this a proper name / comment. This threw me of a couple of times understanding the code.

Reply:
Added comment.